google/cloud/storagebatchoperations/v1/samples/storage_batch_operations_job_samples.cc

```diff
@@ -0,0 +1,199 @@
+    auto const name = parent + "/jobs/" + job_id;
+    auto job = client.GetJob(name);
+    if (!job) throw job.status();
+    std::cout << "Got job: " << job->name() << "\n";
+  }
+  //! [storage_batch_get_job]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void CancelJob(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_cancel_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& project_id, std::string const& location_id,
+     std::string const& job_id) {
+    auto const parent =
+        std::string{"projects/"} + project_id + "/locations/" + "global";
+    auto const name = parent + "/jobs/" + job_id;
+    auto response = client.CancelJob(name);
+    if (!response) throw response.status();
+    std::cout << "Cancelled job: " << name << "\n";
+  }
+  //! [storage_batch_cancel_job]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void DeleteJob(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_delete_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& project_id, std::string const& location_id,
+     std::string const& job_id) {
+    auto const parent =
+        std::string{"projects/"} + project_id + "/locations/" + "global";
+    auto const name = parent + "/jobs/" + job_id;
+    auto status = client.DeleteJob(name);
+    if (!status.ok()) throw status;
+    std::cout << "Deleted job: " << name << "\n";
+  }
+  //! [storage_batch_delete_job]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void AutoRun(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw google::cloud::testing_util::Usage{"auto"};
+  google::cloud::testing_util::CheckEnvironmentVariablesAreSet(
+      {"GOOGLE_CLOUD_PROJECT", "GOOGLE_CLOUD_CPP_STORAGE_TEST_REGION_ID"});
+  auto const project_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value();
+  auto const location_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_STORAGE_TEST_REGION_ID")
+          .value();
+
+  auto gen = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  auto const prefix = std::string{"storage-batch-operations-samples"};
+  auto const job_id =
+      prefix + "_" +
+      google::cloud::internal::Sample(gen, 32, "abcdefghijklmnopqrstuvwxyz");
+
+  auto client =
+      google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient(
+          google::cloud::storagebatchoperations_v1::
+              MakeStorageBatchOperationsConnection());
+
+  std::cout << "\nRunning CreateJob() example\n";
+  CreateJob(client, {project_id, location_id, job_id});
```

Comment:
Since we are using `global` as the location_id, you can remove location_id as a parameter to all of these functions.